### PR TITLE
dotsama parity/orml/simnod/subxt/jsonspree 0.9.36 upgrade #CU-CEN-91

### PR DESCRIPTION
**Pull request by dzmitry-lahoda**
instead of https://github.com/ComposableFi/centauri/pull/201

passes `clear && cargo lcheck  --tests &&  ./scripts/no_std_checks.sh`

### CODE REVIEW

File: algorithms/beefy/prover/Cargo.toml [hunk 1]
```diff
@@ -14,31 +14,37 @@ subxt-codegen = { package = "codegen", path = "../../../utils/subxt/codegen" }
...
-sp-core-hashing = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.33" }
-frame-support = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.33"}
-beefy-mmr = { package = "pallet-beefy-mmr", git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.33" }
-pallet-mmr = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.33" }
-pallet-mmr-primitives = { package = "sp-mmr-primitives", git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.33" }
-beefy-primitives = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.33" }
+sp-std = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.36" }
+sp-runtime = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.36" }
+sp-core =  { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.36" }
+sp-io = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.36", default-features = false }
```
Comment (blasrodri) [inline algorithms/beefy/prover/Cargo.toml:20]: do we need this?

File: hyperspace/Makefile [hunk 2]
```diff
@@ -26,8 +26,8 @@ build-release-hyperspace:
 	cargo b -p $(module-hyperspace) --release
 
 build-docker-image-hyperspace:
-	docker build -f scripts/hyperspace.Dockerfile -t "${composable_centauri_image_with_commit_hash}" .
-	docker tag "${composable_centauri_image_with_commit_hash}" "${composable_centauri_image_latest}"
+	docker build -f scripts/hyperspace.Dockerfile -t latest .
+	docker tag latest
```
Comment (blasrodri) [inline hyperspace/Makefile:30]: why did you remove the possibility of tracking a specific commit?

File: algorithms/beefy/prover/Cargo.toml (hunk 1 above)
Comment (dzmitry-lahoda) [reply algorithms/beefy/prover/Cargo.toml:20]: i did not added any new. only what was before. yes we need changed I done to compile.

File: hyperspace/Makefile (hunk 2 above)
Comment (dzmitry-lahoda) [reply hyperspace/Makefile:30]: sorry. fixing

File: algorithms/beefy/prover/Cargo.toml (hunk 1 above)
Comment (Wizdave97) [inline algorithms/beefy/prover/Cargo.toml:20]:
```suggestion
sp-io = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.36" }
```

Comment (Wizdave97) [reply algorithms/beefy/prover/Cargo.toml:20]: There's no need to turn off default features in the crates here, beefy prover is used only in std environment.

File: algorithms/beefy/prover/Cargo.toml [hunk 3]
```diff
@@ -14,31 +14,37 @@ subxt-codegen = { package = "codegen", path = "../../../utils/subxt/codegen" }
...
-sp-trie = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.33" }
+pallet-mmr-rpc = { package = "mmr-rpc", git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.36" }
+sp-trie = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.36" }
 hex = { version = "0.4.3" }
 beefy-light-client-primitives = {  path = "../primitives" }
 subxt-generated = { path = "../../../utils/subxt/generated" }
 
 [features]
 build-metadata-from-ws = []
+default = ["std"]
```
Comment (Wizdave97) [inline algorithms/beefy/prover/Cargo.toml:45]: I don't think we need this feature flag
```suggestion
```

File: algorithms/beefy/verifier/Cargo.toml [hunk 4]
```diff
@@ -15,34 +15,34 @@ mmr-lib= { package = "ckb-merkle-mountain-range", version = "0.3.2", default-fea
...
+subxt = { git = "https://github.com/paritytech/subxt",  rev = "c0198acea854b24e161dbeb714e71d56830d51ef", features = ["substrate-compat"] }
 hex-literal = "0.3.4"
 serde_json = "1.0.74"
-pallet-mmr-rpc = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.33" }
-frame-support = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.33" }
-sp-io = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.33" }
-sp-trie = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.33" }
+pallet-mmr-rpc = { package = "mmr-gadget", git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.36" }
+frame-support = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.36" }
+sp-io = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.36", default-features = false  }
```
Comment (Wizdave97) [inline algorithms/beefy/verifier/Cargo.toml:40]:
```suggestion
sp-io = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.36"  }
```

**Review by blasrodri — APPROVED** [review]

File: utils/simnode/Cargo.toml [hunk 5]
```diff
@@ -9,23 +9,23 @@ edition = "2021"
 substrate-simnode = { git = "https://github.com/polytope-labs/sc-simnode", rev = "fcfa2dee4ebdd4652375ec9c4b3eba2a63c2d28f" }
```
Comment (Wizdave97) [inline utils/simnode/Cargo.toml:9]: Simnode has a polkadot v0.9.36 release https://github.com/polytope-labs/sc-simnode/tree/polkadot-v0.9.36

**Review by Wizdave97 — CHANGES_REQUESTED** [review]

File: algorithms/beefy/prover/Cargo.toml (hunk 1 above)
Comment (dzmitry-lahoda) [reply algorithms/beefy/prover/Cargo.toml:20]: @andor0 please confirm these are not needed?

File: algorithms/beefy/prover/Cargo.toml (hunk 3 above)
Comment (dzmitry-lahoda) [reply algorithms/beefy/prover/Cargo.toml:45]: @andor0 please confirm these are not needed?

File: algorithms/beefy/verifier/Cargo.toml (hunk 4 above)
Comment (dzmitry-lahoda) [reply algorithms/beefy/verifier/Cargo.toml:40]: i will retain STD as others have STD. if STD needed to be removed (so others would need that too), other PR.

File: algorithms/beefy/prover/Cargo.toml (hunk 1 above)
Comment (obsessed-cake) [reply algorithms/beefy/prover/Cargo.toml]: yes, @Wizdave97 is right.

**Review by obsessed-cake — DISMISSED** [review]

**Review by blasrodri — DISMISSED** [review]

**Review by obsessed-cake — APPROVED** [review]

**Review by Wizdave97 — COMMENTED** [review]
Some unit tests are failing, and it  seems it's required to explicitly add the call index to pallet calls now, check the CI logs.

**Review by Wizdave97 — DISMISSED** [review]
Let's fix the issues mentioned above.

**Comment by dzmitry-lahoda** [comment]
Crazy
```
0s
0s
Run arduino/setup-protoc@v[1](https://github.com/ComposableFi/centauri/actions/runs/4172381765/jobs/7223409324#step:5:1)
/usr/bin/docker exec  8398a6eb9[5](https://github.com/ComposableFi/centauri/actions/runs/4172381765/jobs/7223409324#step:5:6)a8[6](https://github.com/ComposableFi/centauri/actions/runs/4172381765/jobs/7223409324#step:5:7)19d4d1d3ade7f3b95c0aba851446649fa8cd36b725f488a2b5b sh -c "cat /etc/*release | grep ^ID"
Error: API rate limit exceeded for 13.83.14.130. (But here's the good news: Authenticated requests get a higher rate limit. Check out the documentation for more details.)
```
3rd party service used to build arbitrary codesase and use it in relayer

**Comment by obsessed-cake** [comment]
> Crazy
> 
> ```
> 0s
> 0s
> Run arduino/setup-protoc@v[1](https://github.com/ComposableFi/centauri/actions/runs/4172381765/jobs/7223409324#step:5:1)
> /usr/bin/docker exec  8398a6eb9[5](https://github.com/ComposableFi/centauri/actions/runs/4172381765/jobs/7223409324#step:5:6)a8[6](https://github.com/ComposableFi/centauri/actions/runs/4172381765/jobs/7223409324#step:5:7)19d4d1d3ade7f3b95c0aba851446649fa8cd36b725f488a2b5b sh -c "cat /etc/*release | grep ^ID"
> Error: API rate limit exceeded for 13.83.14.130. (But here's the good news: Authenticated requests get a higher rate limit. Check out the documentation for more details.)
> ```
> 
> 3rd party service used to build arbitrary codesase and use it in relayer

yep, I have written about it [here](https://composablefinance.slack.com/archives/C02MSA16DCP/p1674671714470759).

**Comment by dzmitry-lahoda** [comment]
Tests seems passing ok, so need update base branch.

**Review by blasrodri — APPROVED** [review]

**Comment by Wizdave97** [comment]
@dzmitry-lahoda you should have allowed the CI test to complete before merging this.
